app/assets/javascripts/discourse/app/controllers/create-invite.js

```diff
@@ -0,0 +1,147 @@
+import Controller from "@ember/controller";
+import { action } from "@ember/object";
+import { getAbsoluteURL } from "discourse-common/lib/get-url";
+import discourseComputed from "discourse-common/utils/decorators";
+import { ajax } from "discourse/lib/ajax";
+import { extractError } from "discourse/lib/ajax-error";
+import copyText from "discourse/lib/copy-text";
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+import Group from "discourse/models/group";
+import I18n from "I18n";
+
+export default Controller.extend(ModalFunctionality, {
+  onShow() {
+    let inviteKey = "";
+    for (let i = 0; i < 32; ++i) {
+      inviteKey += "0123456789abcdef"[Math.floor(Math.random() * 16)];
```

Comment:
This is confusing to me. Can you explain why we're making an `inviteKey` in the client side?

Reply:
I am generating the invite key on the client side to make it possible to display an invite link even before it is created.

Reply:
How do we handle the case if the ID already exists? I know it seems unlikely but it is possible with this type of GUID generation.

Reply:
I agree, the inviteKey should be generated server side.

I wonder if the `+` icon to create a new invite, can just make either just the invitekey serverside or maybe it could just create an actual invite with all the defaults. And any changes via the modal popup would just be an update to the invite that was just created.

Reply:
I am not so sure about creating an invite when the modal shows up because if a user made a mistake and immediately closes it, then they will have created an invite they did not really need. Maybe I can delete it if the user closes the modal without ever clicking on the create button?